pkcs11/pkcs11.go

```diff
@@ -1590,11 +1590,7 @@ func (s *Slot) generateRSA(o keyOptions) (crypto.PrivateKey, error) {
 	if err != nil {
 		return nil, fmt.Errorf("parsing private key: %w", err)
 	}
-	rsaPriv, ok := priv.(*rsaPrivateKey)
-	if !ok {
-		return nil, fmt.Errorf("expected rsa private key, got: %T", pub)
-	}
-	rsaPriv.WithPublicKeyHandle(pubObj)
+	WithPublicKeyHandle(priv, pubObj)
```

Comment:
Should priv be assigned to the result of this function?

Reply:
Yes, I think you are correct. I've changed it